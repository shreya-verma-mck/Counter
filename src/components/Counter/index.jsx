import React, { useState } from "react";
import "./counter.css";

const Counter = () => {
  const [counter, setCounter] = useState(0);
  return (
    <div className="counterContainer">
      <p className="counterValue">{counter}</p>
      <div className="counterButtonContainer">
        <button
          type="button"
          data-testid="increment"
          className="counterButton"
          onClick={() => setCounter(counter + 1)}
        >
          +
        </button>
        <button
          type="button"
          className="counterButton"
          data-testid="decrement"
          onClick={() => {
            if (counter > 1) {
              setCounter(counter - 1);
            }
          }}
        >
          -
        </button>
      </div>
    </div>
  );
};

export default Counter;
