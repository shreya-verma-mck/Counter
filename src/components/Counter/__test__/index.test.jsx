import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Counter from '..';

describe('Counter', () => {
  it('should show 0 value when it renders initially', () => {
    render(<Counter />);
    expect(screen.queryByText(0)).toBeTruthy();
  });
  it('should increase the counter value by 1 when increment button is clicked', () => {
    render(<Counter />);
    expect(screen.queryByText(1)).toBeFalsy();
    fireEvent.click(screen.getByTestId('increment'));
    expect(screen.queryByText(1)).toBeTruthy();
  });
  it('should decrease the counter value by 1 when decrement button is clicked and the value is greater than 0', () => {
    render(<Counter />);
    fireEvent.click(screen.getByTestId('increment'));
    expect(screen.queryByText(0)).toBeFalsy();
    fireEvent.click(screen.getByTestId('decrement'));
    expect(screen.queryByText(0)).toBeTruthy();
  });

  it('should have the counter value as 0 when decrement button is clicked and the value is 0', () => {
    render(<Counter />);
    expect(screen.queryByText(0)).toBeTruthy();
    fireEvent.click(screen.getByTestId('decrement'));
    expect(screen.queryByText(0)).toBeTruthy();
  });

  it('should render correctly', () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
