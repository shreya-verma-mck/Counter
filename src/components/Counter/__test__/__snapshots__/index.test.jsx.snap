// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly 1`] = `
<DocumentFragment>
  <div
    class="counterContainer"
  >
    <p
      class="counterValue"
    >
      0
    </p>
    <div
      class="counterButtonContainer"
    >
      <button
        class="counterButton"
        data-testid="increment"
        type="button"
      >
        +
      </button>
      <button
        class="counterButton"
        data-testid="decrement"
        type="button"
      >
        -
      </button>
    </div>
  </div>
</DocumentFragment>
`;
